lib/Variable.js

```diff
@@ -14,14 +14,14 @@
  *always* yield true; regardless of the underlying value; because js.
  **/
 
-var check = require('check-types');
+const _ = require('lodash');
 
 function Variable(){
   this.$ = '';
 }
 
 Variable.prototype.set = function( val ){
-  if( !check.nonEmptyString(val) && !check.number(val) && !check.boolean(val) && !check.array(val) && !check.object(val)){
+  if( val === '' || val === null || val === undefined ){
     throw new Error( 'invalid value, value must be valid js Variable' );
```

Comment:
Do we need to check for the falsy number value `NaN` here also?

Reply:
Good question. I was trying to preserve the existing behavior as much and as simply as possible, but `NaN` is definitely in the same boat as `null` and `undefined` as generally not useful.

On the other hand, and especially considering #13, which could be fixed by removing the restriction on storing an empty string, its kinda worth considering if we should filter out any values at all at this point. `null` and `undefined` at least would both cleanly serialize (to nothing) in JSON, so why not allow that if we allow so much else?

Reply:
I'm going to merge this as is but I'm definitely up for re-visiting allowed/disallowed values. I think we should allow `''` to fix #13, figure out if we can stop allowing objects (which just sounds like a painful way to introduce weird behavior), and revisit `null`/`undefined`/`NaN` etc.